test/unit/forge/routes/api/assistant_spec.js

```diff
@@ -92,73 +92,112 @@ describe('Assistant API', async function () {
-            const device = deviceCreateResponse.json()
-            sinon.stub(axios, 'post').resolves({ data: { status: 'ok' } })
+
+    describe('method constraints', async function () {
+        it('should return 400 if method contains capital letters', async function () {
             const response = await app.inject({
                 method: 'POST',
-                url: '/api/v1/assistant/function',
-                headers: { authorization: 'Bearer ' + device.credentials.token },
+                url: '/api/v1/assistant/InvalidMethod',
+                headers: { authorization: 'Bearer ' + TestObjects.tokens.instance },
                 payload: { prompt: 'multiply by 5', transactionId: '1234' }
             })
-            axios.post.calledOnce.should.be.true()
-            response.statusCode.should.equal(200)
+            response.statusCode.should.equal(400)
         })
-        it('instance token can access /function', async function () {
-            sinon.stub(axios, 'post').resolves({ data: { status: 'ok' } })
+        it('should return 400 if method contains invalid characters', async function () {
             const response = await app.inject({
                 method: 'POST',
-                url: '/api/v1/assistant/function',
+                url: '/api/v1/assistant/inv@lid',
                 headers: { authorization: 'Bearer ' + TestObjects.tokens.instance },
                 payload: { prompt: 'multiply by 5', transactionId: '1234' }
             })
-            axios.post.calledOnce.should.be.true()
-            response.statusCode.should.equal(200)
+            response.statusCode.should.equal(400)
         })
-        it('fails when prompt is not supplied', async function () {
+        it('should return 400 if method contains escaped characters', async function () {
             const response = await app.inject({
                 method: 'POST',
-                url: '/api/v1/assistant/function',
+                url: '/api/v1/assistant/method%2Fwith%2Fslashes',
                 headers: { authorization: 'Bearer ' + TestObjects.tokens.instance },
-                payload: { transactionId: '1234' }
+                payload: { prompt: 'multiply by 5', transactionId: '1234' }
             })
             response.statusCode.should.equal(400)
         })
-        it('fails when transactionId is not supplied', async function () {
-            const response = await app.inject({
-                method: 'POST',
-                url: '/api/v1/assistant/function',
-                headers: { authorization: 'Bearer ' + TestObjects.tokens.instance },
-                payload: { prompt: 'multiply by 5' }
+    })
+
+    describe('service tests', async function () {
+        function serviceTests (serviceName) {
+            it('anonymous cannot access', async function () {
+                const response = await app.inject({
+                    method: 'GET',
+                    url: `/api/v1/assistant/${serviceName}n`
```

Comment:
```suggestion
                    url: `/api/v1/assistant/${serviceName}`
```

Typo?

Reply:
yeah, good spot. FYI, it passes because it is a valid param name.